### PR TITLE
DRILL-8054: Add SAS Format Plugin

### PR DESCRIPTION
**Pull request by cgivre**
# [DRILL-8054](https://issues.apache.org/jira/browse/DRILL-8054): Add SAS Format Plugin

## Description
This format plugin adds the ability to read SAS files.  Once merged, Drill will be able to read the same file types as Pandas.

## Documentation
Docs available in README.

## Testing
Added unit tests.

### CODE REVIEW

**Review by luocooong — COMMENTED** [review]
@cgivre @dzamo I did a quick review. Would you mind adding the `reference` lable to this new format plugin?

File: contrib/format-sas/README.md
Comment (dzamo) [inline contrib/format-sas/README.md:5]:
```suggestion
The SAS format supports the `VARCHAR`, `LONG`, `DOUBLE` and `DATE` types.
```

**Review by dzamo — APPROVED** [review]
A new page should appear in the Drill docs shortly

https://drill.apache.org/docs/sas-format-plugin/

File: contrib/format-sas/README.md
Comment (cgivre) [reply contrib/format-sas/README.md:34]: @luocooong Thanks for the review!  Format plugins don't have to have the `enabled` parameter in them.  Simply by including them in the configuration, they are enabled.

File: contrib/format-sas/README.md
Comment (cgivre) [reply contrib/format-sas/README.md:5]: Fixed.

File: contrib/format-sas/src/main/java/org/apache/drill/exec/store/sas/SasFormatConfig.java
Comment (cgivre) [reply contrib/format-sas/src/main/java/org/apache/drill/exec/store/sas/SasFormatConfig.java:51]: Good point.  I lomboked the `toString` as well.

File: contrib/format-sas/src/test/java/org/apache/drill/exec/store/sas/TestSasReader.java
Comment (cgivre) [reply contrib/format-sas/src/test/java/org/apache/drill/exec/store/sas/TestSasReader.java:184]: Fixed

File: contrib/format-sas/src/main/java/org/apache/drill/exec/store/sas/SasBatchReader.java
Comment (cgivre) [reply contrib/format-sas/src/main/java/org/apache/drill/exec/store/sas/SasBatchReader.java:177]: Replaced with `UserException`.

**Comment by cgivre** [comment]
@dzamo @luocooong 
Thank you for the quick review.  I addressed all your review comments.

File: contrib/format-sas/src/main/java/org/apache/drill/exec/store/sas/SasFormatConfig.java
Comment (dzamo) [inline contrib/format-sas/src/main/java/org/apache/drill/exec/store/sas/SasFormatConfig.java:37]: @cgivre toString was one I wasn't sure should be lomboked because we have that `PlanStringBuilder` which I think does some special formatting for display in query plans...

Comment (cgivre) [reply contrib/format-sas/src/main/java/org/apache/drill/exec/store/sas/SasFormatConfig.java:37]: For this case, it doesn't really matter.   It can have a really bad effect on GroupScans and some of the schema planning objects.

Comment (dzamo) [reply contrib/format-sas/src/main/java/org/apache/drill/exec/store/sas/SasFormatConfig.java:35]: Let's merge?

Comment (cgivre) [reply contrib/format-sas/src/main/java/org/apache/drill/exec/store/sas/SasFormatConfig.java:35]: Sure..  I'll wait until the CI passes, then merge.

**Pull request merged**
